Experimental/DAG/luigi_test.py

```diff
@@ -0,0 +1,119 @@
+import os
+import re
+import subprocess
+from glob import glob
+
+import luigi
+
+# Let's define the relationships between the source files here. In practice
+# this information would come from Fab, perhaps it would be stored in
+# something like a sqlite database or similar
+DEPENDENCIES = {
+    "test_exec":
+        ["test_main.f90",],
+    "test_main.f90":
+        ["test.f90",],
+    }
+
+# Similarly another database that gives us information about how to compile
+# each given source. This would likely be more complicated in a final version
+# of Fab; the compiler may be represented as a class or something else
+COMPILERS = {
+    "test_exec": ["ifort",],
+    "test_main.f90": ["ifort", "-c" ],
+    "test.f90": ["ifort", "-c"],    
+    }
```

Comment:
This highlights the issue noted above. `test_exec` is built by `ifort` but `test_main.f90` is not. Rather it is the input **to** `ifort`.

Reply:
Agreed - I expect if this were ever progressed towards a more complete example the `COMPILERS` dictionary wouldn't be shared by both the compilation and linking tasks, which would make the inconsistency there less apparent